Firestore/core/src/firebase/firestore/remote/datastore.h

```diff
@@ -132,6 +137,17 @@ class Datastore : public std::enable_shared_from_this<Datastore> {
   // down.
   bool is_shut_down_ = false;
 
+ protected:
+  /** Test-only method for mocking */
```

Comment:
Note: `Datastore` will have to allow inheritance because of the `MockDatastore` (not ported yet), so it seems that the least intrusive way to add test-only methods would be to make them protected. The tests need access to gRPC completion queue (to be able to pull completions from it) and calls (to be able to cancel them, which is the basis of how `ForceFinish` works).

Reply:
I'm aware about it, though I'm not sure this mechanism would help here. Essentially, I need tests to have access to a couple of private member variables (the comment was misleading, removed the "mocking" part).